### PR TITLE
feat(presenter): new presenter layout and design

### LINKED ISSUE #5: 🚀 [Feature] Layout Presenter Default
## 🚀 Feature

The Layout cannot contain the TopMenu, due to a bug linked in the readme of the presenter.

Layout:
- Section1
- Section2
- Section3
...
- Footer

### LINKED ISSUE #6: 🚀 [Feature] Footer
## 🚀 Feature

image

### LINKED ISSUE #7: 🚀 [Feature] Buttons
## 🚀 Feature

Primary:
image
image
Default/Secondar:
image

Tertiär
image
Quaträr:
image

Size-Varrianten:
Big:
image
Small:
image
Normal:
image

### LINKED ISSUE #8: 🚀 [Feature] Section 1
## 🚀 Feature

image

Remarks:
- Care for Background (Gradient)
- Must integrate with the menu

### LINKED ISSUE #9: 🚀 [Feature] Logo
## 🚀 Feature

Logo Component in different sizes:
image
image
image

### LINKED ISSUE #10: 🚀 [Feature] Link
## 🚀 Feature

Link is like a Button but without animation (current - maybe hover?)

We need it in the menu and the footer
image
image

- [ ] Follow up Issue: Second variant for Links:
image

### LINKED ISSUE #11: 🚀 [Feature] Input Field
## 🚀 Feature

We need an input field with a button(maybe optional as slot)
image

### PR DESCRIPTION
**Pull request by maeckes-infinitylabs**
## 🍰 Pullrequest

Adds new components (with stories and tests) for the presenter frontend.

### Issues

- fixes #5 
- fixes #6
- fixes #7
- fixes #8 
- fixes #9 
- fixes #10 
- fixes #11 

### Todo

- [x] unit tests
- [x] insert correct font family ('poppins')

### CODE REVIEW

File: presenter/src/components/nav/AnchorLink.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    {{ props.label }}
+  </v-btn>
+</template>
+
+<script lang="ts" setup>
+import { computed } from 'vue'
+
+// import { usePageContext } from '#context/usePageContext'
+
+// const pageContext = usePageContext()
```
Comment (ulfgebhardt) [inline presenter/src/components/nav/AnchorLink.vue:13]: remove

File: presenter/src/components/nav/AnchorLink.vue [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// }
+
+const classes = computed(() => ({
+  'anchor-link': true,
+//   'anchor-link--active': isRouteSelected(props.href as string) ? true : "",  
+}))
+
+const style = computed(() => ({
+//   backgroundColor: props.backgroundColor,
+}))
```
Comment (ulfgebhardt) [inline presenter/src/components/nav/AnchorLink.vue:45]: remove

File: presenter/src/components/sections/Section4.vue [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+</template>
+
+<script lang="ts" setup>
+
+</script>
+
+<style scoped lang="scss">
+.section3 {
+  
+}
```
Comment (ulfgebhardt) [inline presenter/src/components/sections/Section4.vue:16]: section4?

File: presenter/src/locales/de.json [hunk 4]
```diff
@@ -40,14 +40,32 @@
...
+    "text4": "Fröhliches Programmieren",
+    "section1":{
+      "headline": "DreamMall created by people for people",
+      "subHeadline": "7 Days free trial",
+      "preOrderBtn": "PRE ORDER"
+    }
   },
   "language": {
     "german": "Deutsch"
   },
```
Comment (ulfgebhardt) [inline presenter/src/locales/de.json:52]: remove

Comment (ulfgebhardt) [reply presenter/src/locales/de.json:52]: Also all the other stuff from the demo

**Review by ulfgebhardt — COMMENTED** [review]
I mostly marked alot of stuff that still needs to go - its not a really helpful review yet.
All in all this looks very good and I am looking forward to it once its finished.

File: presenter/src/components/inputs/MainButton.vue [hunk 5]
```diff
@@ -0,0 +1,93 @@
+<template>
+  <v-btn :class="classes" :style="style" :size="size" @click="onClick">{{ label }}</v-btn>
+</template>
+
+<script lang="ts" setup>
+// eslint-disable-next-line import/no-unassigned-import
```
[the remark this answers is not in the document]
Comment (maeckes-infinitylabs) [reply presenter/src/components/inputs/MainButton.vue]: done

File: presenter/src/components/sections/Section4.vue (hunk 3 above)
Comment (maeckes-infinitylabs) [reply presenter/src/components/sections/Section4.vue]: thx for spotting

File: presenter/src/locales/de.json (hunk 4 above)
Comment (maeckes-infinitylabs) [reply presenter/src/locales/de.json]: done

File: presenter/src/pages/index/index.page.vue [hunk 6]
```diff
@@ -1,19 +1,24 @@
...
-    <p>{{ $t('home.text2') }}</p>
-    <br />
-    <p>{{ $t('home.text3') }}</p>
-    <br />
-    <p>{{ $t('home.text4') }} <v-icon icon="mdi-heart" color="red" /></p>
-    <br />
-    <p>{{ $t('home.greet1') }}</p>
-    <p>{{ $t('home.greet2') }}</p>
+    <div id="section1">
+      <Section1></Section1>
```
Comment (ulfgebhardt) [inline presenter/src/pages/index/index.page.vue:4]:
```suggestion
      <Section1 />
```

File: presenter/src/pages/index/index.page.vue [hunk 7]
```diff
@@ -1,19 +1,24 @@
...
+    </div>
+    <div id="about">
+      <Section2></Section2>
```
Comment (ulfgebhardt) [inline presenter/src/pages/index/index.page.vue:7]:
```suggestion
      <Section2 />
```

File: presenter/src/pages/index/index.page.vue [hunk 8]
```diff
@@ -1,19 +1,24 @@
...
+    </div>
+    <div id="products">
+      <Section3></Section3>
```
Comment (ulfgebhardt) [inline presenter/src/pages/index/index.page.vue:10]:
```suggestion
      <Section3 />
```

File: presenter/src/pages/index/index.page.vue [hunk 9]
```diff
@@ -1,19 +1,24 @@
...
+    </div>
+    <div id="contact">
+      <Section4></Section4>
```
Comment (ulfgebhardt) [inline presenter/src/pages/index/index.page.vue:13]:
```suggestion
      <Section4 />
```

**Review by ulfgebhardt — COMMENTED** [review]
Very clean. besides a few minor issues which are also discussable (annotated) I found those two issues:

1. You test for very specific css in your tests e.g. in `Css tests very specific presenter/src/components/menu/LogoImage.test.ts`.
```
expect(wrapper.find('.v-img__img').attributes().src).toBe('/src/assets/dreammall-logo.svg')
``` 
I'd like to discuss if that is wise and if we can derive a strategy from that. -> I will open an issue for that.

2. German Translation is missing -> I will make an Issue.

I'd like to merge this asap. This would mean we can derive two new tasks from this and have a story book to test tomorrow with Hardi. I Will have a look at the PR locally, maybe I can offer some help.

**Review by ulfgebhardt — APPROVED** [review]

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**